cloudmarker/stores/mongodbstore.py

```diff
@@ -0,0 +1,100 @@
+"""MongoDB store plugin."""
+
+
+import logging
+
+from pymongo import MongoClient
+
+_log = logging.getLogger(__name__)
+
+
+class MongoDBStore:
+    """A plugin to store records on the MongoDB."""
+
+    def __init__(self, host, port, username=None, password=None, db='gcp',
+                 buffer_size=1000):
+        """Initialize object of this class with specified parameters.
+
+        Arguments:
+            host (str): MongoDB instance host.
+            port (int): MongoDB instance port.
+            username (str): MongoDB instance auth username.
+            password (str): MongoDB instance auth password.
+            db (str): MongoDB database name, default: ``gcp``
+            buffer_size (int): Max number of records to hold in memeory for
+                each record_type.
+
+        """
+        # pylint: disable=too-many-instance-attributes
+
+        self._mongodb_host = host
+        self._mongodb_port = port
+        self._mongodb_username = username
+        self._mongodb_password = password
+        self._db = db
+
+        self._buffer_size = buffer_size
+        self._buffer = {}
+
+        if not (username and password):
+            self._client = MongoClient(host=self._mongodb_host,
+                                       port=self._mongodb_port,
+                                       connect=False)
+        else:
+            self._client = MongoClient(host=self._mongodb_host,
+                                       port=self._mongodb_port,
+                                       username=username,
+                                       password=password,
+                                       authSource=self._db,
+                                       authMechanism='SCRAM-SHA-256',
+                                       connect=False)
+
+    def _flush(self, record_type, records):
```

Comment:
Is the second param really required, since we already have a reference to the `self._buffer` dict. The `record` param is just a pointer to the actual array, I think it can be simplified by just doing a lookup from the buffer.